addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-service.js

```diff
@@ -210,26 +222,16 @@ class EnvironmentScService extends Service {
     const sagemakerUpdated = {};
     _.forEach(sagemakerInstances, async (existingEnvRecord, key) => {
       const expectedDDBStatus = SageMakerStatusMap[sagemakerRealtimeStatus[key]];
+      let updateStatusResult;
       if (expectedDDBStatus && existingEnvRecord.status !== expectedDDBStatus) {
-        const newEnvironment = {
-          id: existingEnvRecord.id,
-          rev: existingEnvRecord.rev || 0,
-          status: SageMakerStatusMap[sagemakerRealtimeStatus[key]].toUpperCase(),
-        };
-        try {
-          // Might run into situation where the environment was just updated and rev number does not match
-          // Log the error and skip the update for now
-          // The next invocation of poll and sync will do the sync if it's still needed
-          await this.update(requestContext, newEnvironment);
-          sagemakerUpdated[key] = {
-            ddbID: existingEnvRecord.id,
-            currentStatus: expectedDDBStatus,
-            staleStatus: existingEnvRecord.status,
-          };
-        } catch (e) {
-          this.log.error(`Error updating record ${existingEnvRecord.id}`);
-          this.log.error(e);
-        }
+        updateStatusResult = await this.updateDDBStatus(requestContext, existingEnvRecord, expectedDDBStatus);
```

Comment:
Looks like there is an opportunity to DRY the code between here and the `pollAndSyncEc2Status` method.